File: packages/case-boundary/src/mappers/LogPrinter.ts
import { TestPrinter } from '@contract-case/case-core';
import { ILogPrinter, IResultPrinter } from '../boundary';
import { handleVoidResult } from './Result';

export const wrapLogPrinter = (
  externalLogger: ILogPrinter,
  externalPrinter: IResultPrinter
): TestPrinter => ({
  log: (
    level: string,
    timestamp: string,
    version: string,
    typeString: string,
    location: string,
    message: string,
    additional: string
  ) =>
    handleVoidResult(
      externalLogger.log(
        level,
        timestamp,
        version,
        typeString,
        location,
        message,
        additional
      ),
      'CaseCoreError'
    ),

  printMatchError: (details) =>
    handleVoidResult(externalPrinter.printMatchError(details), 'CaseCoreError'),
  printMessageError: (details) =>
    handleVoidResult(
      externalPrinter.printMessageError(details),
      'CaseCoreError'
    ),
  printTestTitle: (details) =>
    handleVoidResult(externalPrinter.printTestTitle(details), 'CaseCoreError'),
});

File: packages/dsl-java/src/main/java/io/contract_testing/contractcase/logs/LogPrinter.java
package io.contract_testing.contractcase.logs;

import io.contract_testing.contractcase.logs.PrintableMatchError;
import io.contract_testing.contractcase.logs.PrintableMessageError;
import io.contract_testing.contractcase.logs.PrintableTestTitle;
import org.jetbrains.annotations.NotNull;

/**
 * Interface for printing logs and test results
 */
public interface LogPrinter {

  void log(@NotNull String level, @NotNull String timestamp,
      @NotNull String version, @NotNull String typeString, @NotNull String location,
      @NotNull String message, @NotNull String additional);

  void printMatchError(
      @NotNull PrintableMatchError description);

  void printMessageError(
      @NotNull PrintableMessageError description);

  void printTestTitle(@NotNull PrintableTestTitle titleDetails);
}
